addons/calendar/calendar.py

```diff
@@ -1449,14 +1459,24 @@ def get_interval(self, cr, uid, ids, date, interval, tz=None, context=None):
             date = date.replace(tzinfo=pytz.timezone('UTC')).astimezone(timezone)
 
         if interval == 'day':
-            res = str(date.day)
+            # Day number (1-31)
+            res = unicode(date.day)
+
         elif interval == 'month':
-            res = date.strftime('%B') + " " + str(date.year)
+            # Localized month name and year
+            res = babel.dates.format_date(date=date, format='MMMM Y', locale=context.get('lang', 'en_US'))
```

Comment:
I think we need to use a lowercase `y`, i.e. `MMMM y`, because uppercase `Y` is [the ISO year](http://babel.edgewall.org/wiki/Documentation/dates.html#date-fields), and should only be used in combination with the ISO week number `w`. For example `Y==2005` for 2006-1-1, because that day fell into ISO week 53 of year 2005. See this [commit message](https://github.com/odoo/odoo/commit/c0644a547444a5c33e5b9c7e2e55f7f45a2889b7) for some extra info.